ethereum_history_api/circuits/lib/src/merkle_patricia_proofs/proof_test.nr

```diff
@@ -451,4 +503,52 @@ mod verify_merkle_proof {
             cancun_transaction_proof_input.proof
         );
     }
+
+    #[test(should_fail_with="Key mismatch in the leaf")]
+    fn fail_invalid_key() {
+        let mut invalid_key = london_state_proof_input.key;
+        invalid_key[invalid_key.len() - 1] += 1;
+        verify_merkle_proof(
+            invalid_key,
+            london_state_proof_input.value,
+            london_state_root,
+            london_state_proof_input.proof
+        );
+    }
+
+    #[test(should_fail_with="Value mismatch")]
+    fn fail_invalid_value() {
+        let mut invalid_value = london_state_proof_input.value;
+        invalid_value[invalid_value.len() - 1] += 1;
+        verify_merkle_proof(
+            london_state_proof_input.key,
+            invalid_value,
+            london_state_root,
+            london_state_proof_input.proof
+        );
+    }
+
+    #[test(should_fail_with="Invalid node hash")]
+    fn fail_invalid_root() {
+        let mut invalid_root = london_state_root;
+        invalid_root[0] += 1;
+        verify_merkle_proof(
+            london_state_proof_input.key,
+            london_state_proof_input.value,
+            invalid_root,
+            london_state_proof_input.proof
+        );
+    }
+
+    #[test(should_fail_with="Invalid node hash")]
+    fn fail_invalid_proof() {
+        let mut invalid_proof = london_state_proof_input.proof;
+        invalid_proof.nodes[invalid_proof.depth - 2][0] -= 1;
```

Comment:
Can we do alter_array here?

Reply:
Not really, because we need to subtract one instead of one for the compilator not to panic